libavcodec/decode.c

```diff
@@ -1202,6 +1229,15 @@ int ff_get_format(AVCodecContext *avctx, const enum AVPixelFormat *fmt)
         return AV_PIX_FMT_NONE;
 
     for (;;) {
+        if (avctx->internal->hwaccel_priv_data &&
+            avctx->hwaccel->caps_internal & HWACCEL_CAP_INTERNAL_ALLOC) {
+            err = hwaccel_realloc_surface(avctx);
```

Comment:
Is a flush needed before realllocating surface ?

Reply:
> Is a flush needed before realllocating surface ?

The previous surface and hw_frames_ctx is stored/referenced by each AVFrame. So just need to create new hw_frames_ctx, and the new AVFrame will pick up surface from it. The old AVFrame still keep there with old surface if it is put into reference list. So don't need to flush it.